### PR TITLE
Feat/44 add crud functionality

### LINKED ISSUE #44: Add CRUD functionality
Implement the following CRUD functions using the database to various pages:

- [x] (Create) functionality to the recipe contribute page to add new recipes
- [x] (Read) functionality to the home page to populate information on recipe cards
- [x] (Read) functionality to the recipe details page to show recipe information
- [x] (Update) functionality to the recipe contribute page to edit recipes
- [x] (Delete) functionality to the recipe details page to remove a recipe

### PR DESCRIPTION
**Pull request by edbertdai**
Implement initial CRUD functionality using database.
The following pages are fully functional:
- Homepage
- Recipe details (with delete function)
- Recipe contribute

A `Database` class is added that contains functions for writing, modifying, and deleting from the database.

Closes #44

### CODE REVIEW

File: source/components/home-page/home-page.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//import "@fortawesome/fontawesome-free/js/all.js";
```
Comment (jbutera94) [inline source/components/home-page/home-page.js:1]: Please remove this line if not used.

File: source/components/recipe-contribute/recipe-contribute.js [hunk 2]
```diff
@@ -7,30 +10,271 @@ class RecipeContribute extends HTMLElement {
...
+      },
+      nutrients: {
+        totalServings: 0,
+        calories: 0,
+        protein: "",
+        fat: "",
+      },
+      description: "",
+      ingredients: [],
+      steps: "<ol>",
```
Comment (jbutera94) [inline source/components/recipe-contribute/recipe-contribute.js:88]: Why does this start with `<ol>`? Should it be blank instead?

File: source/config.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "firebaseConfig": {
+    "apiKey": "",
```
Comment (jbutera94) [inline source/config.json:3]: I forget, were we gonna push this?

**Review by jbutera94 — CHANGES_REQUESTED** [review]
Looks good overall, left a few suggestions.

Also please replace line 1 of router.js with the following block:
```javascript
if (typeof (module) !== 'undefined' && module.exports) {
    module.exports = { getRoutefromUrl };
}
```

File: source/components/recipe-contribute/recipe-contribute.js (hunk 2 above)
Comment (edbertdai) [reply source/components/recipe-contribute/recipe-contribute.js:88]: It starts with <ol> because that's what steps and I append stuff afterwards.

Comment (edbertdai) [reply source/components/recipe-contribute/recipe-contribute.js:88]: But I guess I'll just leave it blank then.

File: source/components/home-page/home-page.js (hunk 1 above)
Comment (edbertdai) [reply source/components/home-page/home-page.js:1]: I don't think it's used but it was there and commented when I started.

Comment (edbertdai) [reply source/components/home-page/home-page.js:1]: I'll remove it then.

File: source/components/recipe-contribute/recipe-contribute.js (hunk 2 above)
Comment (jbutera94) [reply source/components/recipe-contribute/recipe-contribute.js:88]: Oh then nevermind, you can leave it the way it is

Comment (edbertdai) [reply source/components/recipe-contribute/recipe-contribute.js:88]: I already changed it but I think you're right. It makes the code more concise that way.

File: source/config.json (hunk 3 above)
Comment (edbertdai) [reply source/config.json]: l33t h4x0rz

**Review by jbutera94 — APPROVED** [review]

**Pull request merged**